horace_core/sqw/file_io/@sqw_binfile_common/get_pix_in_ranges.m

```diff
@@ -35,46 +35,62 @@
 [pix_starts, pix_ends] = merge_adjacent_ranges(pix_starts, pix_ends);
 
 % Position file reader at start of first block of pixels to read
-first_seek_pos = obj.pix_pos_ + (pix_starts(1) - 1)*PIXEL_SIZE;
-do_fseek(obj.file_id_, first_seek_pos, 'bof');
+%first_seek_pos = obj.pix_pos_ + (pix_starts(1) - 1)*PIXEL_SIZE;
+%do_fseek(obj.file_id_, first_seek_pos, 'bof');
 
-blocks = cell(1, numel(pix_starts));
-for i = 1:numel(pix_starts)
-    num_pix_to_read = pix_ends(i) - pix_starts(i) + 1;
-    read_size = [PixelData.DEFAULT_NUM_PIX_FIELDS, num_pix_to_read];
-    blocks{i} = fread_catch(obj.file_id_, read_size, '*float32');
-
-    try
-        seek_size = (pix_starts(i + 1) - pix_ends(i) - 1)*PIXEL_SIZE;
-    catch ME
-        if strcmpi(ME.identifier, 'MATLAB:badsubscript')
-            % we've read in the final block, no more seeking to do
-            break
-        end
-    end
-    do_fseek(obj.file_id_, seek_size, 'cof');
-end
+% blocks = cell(1, numel(pix_starts));
+% for i = 1:numel(pix_starts)
+%     seek_pos = obj.pix_pos_ + (pix_starts(i) - 1)*PIXEL_SIZE;
+%     do_fseek(obj.file_id_, seek_pos, 'bof');
+%     num_pix_to_read = pix_ends(i) - pix_starts(i) + 1;
+%
+%     read_size = [PixelData.DEFAULT_NUM_PIX_FIELDS, num_pix_to_read];
+%     %blocks{i} = fread_catch(obj.file_id_, read_size, '*float32');
+%     blocks{i} = fread(obj.file_id_,read_size, '*float32');
+%
+%     try
+%         seek_size = (pix_starts(i + 1) - pix_ends(i) - 1)*PIXEL_SIZE;
+%     catch ME
+%         if strcmpi(ME.identifier, 'MATLAB:badsubscript')
+%             % we've read in the final block, no more seeking to do
+%             break
+%         end
+%     end
+%     do_fseek(obj.file_id_, seek_size, 'cof');
+%end
+blocks = arrayfun(@(pix_start,pix_end)(read_block(obj,pix_start,pix_end)),...
+    pix_starts,pix_ends,'UniformOutput',false);
 pix = [blocks{:}];
 
 end  % function
 
+function block = read_block(obj,pix_start,pix_end)
+seek_pos = obj.pix_pos_ + (pix_start - 1)*obj.pixel_size;
+do_fseek(obj.file_id_, seek_pos, 'bof');
+read_size = [9,pix_end-pix_start+1];
```

Comment:
Hard-coding the number of fields here is not a good idea - maybe you can pass `PixelData.DEFAULT_NUM_PIX_FIELDS` to the `read_block` function instead?

Reply:
This is the issue I spent some time thinking about and after some hesitation come to this conclusion. A static global variable `PixelData.DEFAULT_NUM_PIX_FIELDS` is convenient in the situation where you want to change many things in one place. This is not our situation, as if you want to add support for other pixel size (which I think we will want to do one day -- at least to try -- I may want to discuss the merit of this job in a future, but here I want to make it most convenient in to do if decided for it) you need to: `a)` keep support for previous size for compatibility. `b)` overload the number of functions to support different pixel size. I found it more convenient to see the consequences of overloading if the pixel size is hardcoded. It become immediately obvious what functions return different result

Reply:
I am not entirely sure in this conclusion. The code left commented for further review, analysis and modifications

Reply:
I've added the ticket (#686) and removed some hard-coding. Not sure in the usefulness, as read operation has hardcoding anyway (`float32*`). But ticket should resolve the issues